### PR TITLE
Added Assets for Web Browser MP Script

### PR DESCRIPTION
**Pull request by cain-kilgore**
## Test Plan
1. Run the Script: `http://mpassets.highfidelity.com/41f09ec3-976c-49f2-89fe-2d0e1a2b4a98-v1/webbrowser.js`
2. Confirm that you see a new Web Browser app in your Toolbar and Tablet
3. Confirm that it automatically loads https://highfidelity.com
4. Confirm that you can browse to different addresses using the top bar. 

This script will add a web browser which you can use directly from your toolbar or tablet in High Fidelity. This PR has been created due to the inability to add QML's to Marketplace Applications.

Ready for testing.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2430/HighFidelity-Beta-PR11170-4e81ebf85cafce1d7deebac1311dc110bbe4d404.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2430/HighFidelity-Beta-PR11170-4e81ebf85cafce1d7deebac1311dc110bbe4d404.exe

**Comment by Nex-Pro** [comment]
Looking at this.

**Comment by Nex-Pro** [comment]
Test Plan 

Run the Script: http://mpassets.highfidelity.com/41f09ec3-976c-49f2-89fe-2d0e1a2b4a98-v1/webbrowser.js
Confirm that you see a new Web Browser app in your Toolbar and Tablet **SUCCESS**
Confirm that it automatically loads https://highfidelity.com **SUCCESS**
Confirm that you can browse to different addresses using the top bar. **SUCCESS**

This script will add a web browser which you can use directly from your toolbar or tablet in High Fidelity. This PR has been created due to the inability to add QML's to Marketplace Applications

@cain-kilgore  for some reason this build does not give me the option to use HMD.  can you double check please. HMD is fine in dev #7008 etc. have started interface several times.

**Comment by cain-kilgore** [comment]
@Nex-Pro Yeah, it started happening after Dev Build 7001 -  a bad PR got merged in and broke HMD. Not related to this PR. :)

File: interface/resources/qml/controls-uit/WebGlyphButton.qml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//
+//  GlyphButton.qml
+//
+//  Created by Vlad Stelmahovsky on 2017-06-21
```
Comment (druiz17) [inline interface/resources/qml/controls-uit/WebGlyphButton.qml]: If you actually created these file please put you name as the creator

File: interface/resources/qml/hifi/WebBrowser.qml [hunk 2]
```diff
@@ -0,0 +1,253 @@
+//
+//  WebBrowser.qml
+//
+//
+//  Created by Vlad Stelmahovsky on 06/22/2017
```
Comment (druiz17) [inline interface/resources/qml/hifi/WebBrowser.qml]: Change creator name to yours if you actually created this file.

Comment (cain-kilgore) [inline interface/resources/qml/hifi/WebBrowser.qml]: I did not @druiz17

File: interface/resources/qml/controls-uit/WebGlyphButton.qml (hunk 1 above)
Comment (cain-kilgore) [inline interface/resources/qml/controls-uit/WebGlyphButton.qml]: I did not - it was @vladest who created these. My task was to prep these for a Marketplace script.

File: interface/resources/qml/hifi/WebBrowser.qml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+import "../controls"
+
+Rectangle {
+    id: root;
+
+    HifiConstants { id: hifi; }
+
+    property string title: "";
+    signal sendToScript(var message);
+    property bool keyboardEnabled: true  // FIXME - Keyboard HMD only: Default to false
```
Comment (zfox23) [inline interface/resources/qml/hifi/WebBrowser.qml]: Fixme?

**Comment by zfox23** [comment]
I had a review pending when this was merged. I have some problems with this browser (the code seems fine, besides one "FIXME" about whose presence I'm uncertain - although, with browsers, it's hard to know if it's bulletproof :)) Here are the comments I was going to make:

I should be able to type in "google.com" into the address bar. Right now, I can't, which feels like a blocker to me:

image

Do we also want to enable users to be able to Google search by typing in text into the URL bar?

Also, it doesn't look like this browser supports multiple windows or popups, making many sites unusable. Is this OK?

**Comment by zfox23** [comment]
I'd like @cain-kilgore or @vladest to address my concerns, despite this being merged already.

**Comment by changxc** [comment]
@zfox23 @vladest @cain-kilgore  

(A) Address bar
If my understanding is correct, full URL typing works (https://www.google.com/) but typing incomplete URL (google.com) will result in error message (This webpage is not found ERR_FILE_NOT_FOUND)

Options:
(1) <new feature request> when user type in incomplete URL, leverage google or other search engine to show the most relevant web search result 
(2)  <quick fix but less ideal> display error message that instruct user to type complete URL instead+ make google.com the homepage instead of highfidelity.com

I would vote for option (1) 

(B) Pop-up
I am not strong on supporting pop-up. I saw other cases where built-in-app browsers not supporting pop-ups. It is not the best user experience but we are also not promising a full browser in HF. 
Thoughts?

**Comment by zfox23** [comment]
@changxc - thanks for your reply! I agree with your assessment of (A).

(B): I am concerned about the lack of popup support because so many sites rely on them. For example, many websites that require a login will pop up an authentication page when using external services (like Google signin, Facebook signin, etc). Additionally, any site whose links don't use the current page as its target (i.e. a majority of blogs) will not function properly.

As a model, I think we should look at Steam's in-game browser. It's _very_ barebones, but still supports features like tabbed browsing and popups.

**Comment by changxc** [comment]
Sounds like a plan. I will add the additional feature request to FB, which is currently assigned to @vladest  https://highfidelity.fogbugz.com/f/cases/7001/Tablet-Web-Browser-issues-after-PR-merge

(A) when user type in incomplete URL, leverage google or other search engine to show the most relevant web search result

(B) support pop-up (especially authentication page) 

Thank you!

**Comment by cain-kilgore** [comment]
@changxc Will I need to take any action in this case?

**Comment by changxc** [comment]
@cain-kilgore not at this moment, thank you again!

**Comment by changxc** [comment]
thank you  - now we can finally merge it

**Pull request merged**